### PR TITLE
Web fixes

### PR DESCRIPTION
**Pull request by gwsii**
Numerous web fixes and merge into develop.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -16,6 +16,7 @@
     "writer:deploy":        "./build/0.env.sh && ./build/2.sls.sh deploy \"Deploying Event Writer\"                          ./event-writer",
     "writer:remove":        "./build/0.env.sh && ./build/2.sls.sh remove \"Removing Event Writer\"                           ./event-writer",
 
+    "stream:install":       "./build/0.env.sh && ./build/1.install.sh \"Installing Retail Stream Dependencies\"               ./retail-stream",
```
Comment (laurwang) [inline package.json]: Should be removed.  All node-module-needing code has been moved out of stream to writer.

File: web/components/login/amazon-login.jsx [hunk 2]
```diff
@@ -132,11 +136,11 @@ class AmazonLogin extends Component {
   }
 
   sendUserLogin() {
-    this.makeApiRequest(config.EventWriterApi, 'POST', '/event-writer/', {
+    this.makeApiRequest(config.RetailStreamAPI, 'POST', '/event-writer/', {
```
Comment (laurwang) [inline web/components/login/amazon-login.jsx]: Should be develop version (calling EventWriterApi)

File: web/serverless.yml [hunk 3]
```diff
@@ -23,6 +23,9 @@ custom:
       - HelloRetailWebDistribution
       - DomainName
 
+  # Hosted Zone ID for all of CloudFormation.
+  cloudFrontHostedZoneId: Z2FDTNDATAQYW2
```
Comment (laurwang) [inline web/serverless.yml]: Please remove cloudFrontHostedZoneId.

File: web/serverless.yml [hunk 4]
```diff
@@ -237,12 +240,12 @@ resources:
           Type: A
           AliasTarget:
             DNSName: ${self:custom.cloudFrontDomainName}
-            HostedZoneId: ${self:custom.hostedZoneId}
+            HostedZoneId: ${self:custom.cloudFrontHostedZoneId}
```
Comment (laurwang) [inline web/serverless.yml]: All occurrences of cloudFrontHostedZoneId should be hostedZoneId

File: web/serverless.yml [hunk 5]
```diff
@@ -237,12 +240,12 @@ resources:
...
         - Name: ${self:custom.dnsHostName}
           Type: AAAA
           AliasTarget:
             DNSName: ${self:custom.cloudFrontDomainName}
-            HostedZoneId: ${self:custom.hostedZoneId}
+            HostedZoneId: ${self:custom.cloudFrontHostedZoneId}
```
Comment (laurwang) [inline web/serverless.yml]: All occurrences of cloudFrontHostedZoneId should be hostedZoneId

File: web/serverless.yml (hunk 3 above)
Comment (gwsii) [inline web/serverless.yml]: Why do you want this removed? Have you tried to deploy the web stack without it?

**Comment by laurwang** [comment]
Yes, deployed with all instances of cloudFrontHostedZoneId changed to hostedZoneId, which is already in (my) private.yml, with that Id.

File: web/serverless.yml (hunk 3 above)
Comment (gwsii) [inline web/serverless.yml]: Per our discussion, this zone ID is globally known and used by all CloudFront hosted websites. No need to keep secret.

File: web/serverless.yml [hunk 6]
```diff
@@ -33,9 +36,6 @@ custom:
       - ${self:custom.domainName}
       - ${self:custom.stage}.${self:custom.domainName}
 
-  # Hosted Zone ID for website DNS.
-  hostedZoneId: ${self:custom.private.hostedZoneId}
-
```
Comment (gwsii) [inline web/serverless.yml]: And since this is not used anywhere, it can be removed.

**Comment by gwsii** [comment]
Hi @laurwang! Thanks for the review and the offline discussion. :D Think I have resolved your feedback, to please have another look. Thanks! ~Greg

**Pull request merged**